java/vector/src/test/java/org/apache/arrow/vector/ipc/TestArrowFile.java

```diff
@@ -708,6 +710,25 @@ public void testWriteReadVarBin() throws IOException {
     }
   }
 
+  @Test
+  public void testReadWriteMultipleBatches() throws IOException {
+    File file = new File("target/mytest_nulls_multibatch.arrow");
+
+    try (IntVector vector = new IntVector("foo", allocator);) {
+      Schema schema = new Schema(Collections.singletonList(vector.getField()), null);
+      try (FileOutputStream fileOutputStream = new FileOutputStream(file);
+           VectorSchemaRoot root = new VectorSchemaRoot(schema, Collections.singletonList((FieldVector) vector), vector.getValueCount());
+           ArrowFileWriter writer = new ArrowFileWriter(root, new MapDictionaryProvider(), fileOutputStream.getChannel());) {
```

Comment:
the second argument `provider` is usually null if there is no Dictionary being used